### PR TITLE
Provide a better explanation of the `max_change` and `min_change` of `TimeStepWizard`

### LINKED ISSUE #3244: Add to the docs a better explanation about `max_change` and `min_change` of the `TimeStepWizard`
Hi,

I have been confused about what the `max_change` and `min_change` means.
I understand from the documentation that this is to avoid the time step to change too quickly, but how is it used?

Is it in time units? e.g., taking the default `max_change` of 1.1, that means that for dated simulation it will change the time step at max in 1.1 seconds? 

Is it a multiplication? e.g., `new_time_step = 1.1*time_step` ?

I think we can try to be more clear about that in the docs.

### PR DESCRIPTION
**Pull request by iuryt**
closes #3244

### CODE REVIEW

File: src/Simulations/time_step_wizard.jl [hunk 1]
```diff
@@ -33,7 +33,9 @@ Base.summary(wizard::TimeStepWizard) = string("TimeStepWizard(",
 Callback for adapting simulation to maintain the advective Courant-Freidrichs-Lewy (CFL)
 number to `cfl`, the `diffusive_cfl`, while also maintaining `max_Δt`, `min_Δt`, and
 satisfying `max_change` and `min_change` criteria so that the simulation's timestep
-`simulation.Δt` is not adapted "too quickly".
+`simulation.Δt` is not adapted "too quickly". In other words, `max_change` is the maximum
+and `min_change` is the minimum relative change of the time step and are therefore non-dimensional
+(`min_change * old_Δt <= new_Δt <= max_change * old_Δt`).
```
Comment (navidcy) [inline src/Simulations/time_step_wizard.jl:38]:
```suggestion
and `min_change` is the minimum relative change of the time step and are, therefore, non-dimensional
(`min_change * old_Δt ≤ new_Δt ≤ max_change * old_Δt`).
```

**Review by navidcy — APPROVED** [review]

File: examples/horizontal_convection.jl [hunk 2]
```diff
@@ -108,7 +108,7 @@ simulation = Simulation(model, Δt=1e-2, stop_time=40.0)
 #
 # The `TimeStepWizard` manages the time-step adaptively, keeping the Courant-Freidrichs-Lewy 
 # (CFL) number close to `0.75` while ensuring the time-step does not increase beyond the 
-# maximum allowable value for numerical stability.
+# maximum allowable value for numerical stability (`new_Δt <= max_change * old_Δt`).
```
Comment (navidcy) [inline examples/horizontal_convection.jl:111]:
```suggestion
# maximum allowable value for numerical stability (`new_Δt ≤ max_change * old_Δt`).
```

Comment (navidcy) [reply examples/horizontal_convection.jl:111]: Actually the phrasing here with the parenthesis introduced suggests that `max_change * old_Δt` is what numerical stability requires. But that's not true.

File: examples/horizontal_convection.jl [hunk 3]
```diff
@@ -108,7 +108,7 @@ simulation = Simulation(model, Δt=1e-2, stop_time=40.0)
 #
 # The `TimeStepWizard` manages the time-step adaptively, keeping the Courant-Freidrichs-Lewy 
 # (CFL) number close to `0.75` while ensuring the time-step does not increase beyond the 
-# maximum allowable value for numerical stability.
+# maximum allowable value for numerical stability (`new_Δt ≤ max_change * old_Δt`).
```
Comment (iuryt) [inline examples/horizontal_convection.jl:111]:
```suggestion
# maximum allowable value for numerical stability while avoiding the time-step to change more than
# the predefined maximum change (`new_Δt ≤ max_change * old_Δt`).
```

**Review by navidcy — APPROVED** [review]

File: src/Simulations/time_step_wizard.jl [hunk 4]
```diff
@@ -33,7 +33,9 @@ Base.summary(wizard::TimeStepWizard) = string("TimeStepWizard(",
 Callback for adapting simulation to maintain the advective Courant-Freidrichs-Lewy (CFL)
 number to `cfl`, the `diffusive_cfl`, while also maintaining `max_Δt`, `min_Δt`, and
 satisfying `max_change` and `min_change` criteria so that the simulation's timestep
-`simulation.Δt` is not adapted "too quickly".
+`simulation.Δt` is not adapted "too quickly". In other words, `max_change` is the maximum
+and `min_change` is the minimum relative change of the time step and are, therefore, non-dimensional
+(`min_change * old_Δt ≤ new_Δt ≤ max_change * old_Δt`).
```
Comment (glwagner) [inline src/Simulations/time_step_wizard.jl:38]: Note there is a grammatical error in the above as well. If we're seeking improved clarity, I suggest changing to something like

```julia
Callback function that adapts the simulation time step to a target
advective Courant-Freidrichs-Lewy (CFL) number `cfl`, within the limits

    max(max_Δt, max_change * previous_Δt) ≤ new_Δt ≤ min(min_Δt, min_change * previous_Δt)

where `new_Δt` is the new time step calculated by `TimeStepWizard`.
```

Or something along those lines...

**Pull request merged**